imblearn/base.py

```diff
@@ -78,12 +80,20 @@ def fit_resample(self, X, y):
 
         output = self._fit_resample(X, y)
 
+        if self._columns is not None:
+            import pandas as pd
+            X_ = pd.DataFrame(output[0], columns=self._columns)
+        else:
+            X_ = output[0]
+
```

Comment:
Why not doing this in `_check_X_y`?

Reply:
This block required the resampled data so `_check_X_y` is called before that the resampling happened.

Reply:
Ok. It is like we need to extracted it in an internal transformation method. But I am ok as is.